Lib/test/test_pyrepl/test_reader.py

```diff
@@ -358,6 +359,59 @@ def test_setpos_from_xy_for_non_printing_char(self):
         reader.setpos_from_xy(8, 0)
         self.assertEqual(reader.pos, 7)
 
+    def test_empty_line_control_w_k(self):
+        """Test that Control-W followed by Control-K on an empty line doesn't crash."""
+        events = itertools.chain(
+            [
+                Event(evt="key", data="\x17", raw=bytearray(b"\x17")),  # Control-W
+                Event(evt="key", data="\x0b", raw=bytearray(b"\x0b")),  # Control-K
+            ],
+        )
```

Comment:
You can simplify this:
```suggestion
        events = [
            Event(evt="key", data="\x17", raw=bytearray(b"\x17")),  # Control-W
            Event(evt="key", data="\x0b", raw=bytearray(b"\x0b")),  # Control-K
        ]
```